tests/commonTemplates_test.go

```diff
@@ -212,4 +215,83 @@ var _ = Describe("Common templates", func() {
 			table.Entry("[test_id:4770]golden image NS", &goldenImageNS),
 		)
 	})
+
+	Context("older templates update", func() {
+		It("should replace ownerReference with owner annotations for older templates", func() {
+			ssp := getSsp()
+
+			// Create a dummy template to act as an owner for the test template
+			ownerTemplate := func() *templatev1.Template {
+				tpl := &templatev1.Template{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "owner-template",
+						Namespace: ssp.Spec.CommonTemplates.Namespace,
+					},
+				}
+				Expect(apiClient.Create(ctx, tpl)).ToNot(HaveOccurred(), "failed to create dummy owner for an old template")
+				// key, err := client.ObjectKeyFromObject(tpl)
+				// Expect(err).ToNot(HaveOccurred(), "failed to read template object key")
+				// Expect(apiClient.Get(ctx, key, tpl)).ToNot(HaveOccurred())
+
+				return tpl
+			}()
+
+			oldTpl := func() *templatev1.Template {
+				tpl := &templatev1.Template{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "test-old-template",
+						Namespace: ssp.Spec.CommonTemplates.Namespace,
+						Labels: map[string]string{
+							"template.kubevirt.io/version": "not-latest",
+							"template.kubevirt.io/type":    "base",
+						},
+						OwnerReferences: []v1.OwnerReference{{
+							APIVersion: "template.openshift.io/v1",
+							Kind:       "Template",
+							Name:       ownerTemplate.Name,
+							UID:        ownerTemplate.UID,
+						}},
+					},
+				}
+				Expect(apiClient.Create(ctx, tpl)).ToNot(HaveOccurred(), "creation of dummy old template failed")
+
+				return tpl
+			}()
+
+			// Add dummy annotation to SSP resource to initiate reconciliation
+			updateSsp(func(foundSsp *sspv1beta1.SSP) {
+				if foundSsp.GetAnnotations() == nil {
+					foundSsp.SetAnnotations(map[string]string{})
+				}
+
+				foundSsp.GetAnnotations()["dummy"] = "dummy"
+			})
+
```

Comment:
The part in this `It` block till now seems good to be moved to a `BeforeEach` for this context
All the tests in this `Context` are going to need a dummy template and a reconciliation trigger

Reply:
I didn't want to add `BeforeEach` and `AfterEach` until they are needed, in this case there is a single test in the `Context` so it is not needed, but it does make sense to make it more future proof, I will update

Reply:
Done